frontend/src/app/gitImport.tsx

```diff
@@ -97,35 +98,40 @@ class GitImport extends Component {
         'cache-control': 'no-cache'
       }
 
-      fetch('/content/repositories/GIT_INTEGRATION_URL')
+      fetch('/content/repositories/pantGitServiceURL')
         .then((resp) => resp.text()).then((text) => {
           this.setState({ git2pantheonURL: text })
         }).then(() => {
-          console.log("The git2pantheon URL is: " + this.state.git2pantheonURL)
-          const payload = {
-            branch: this.state.branch,
-            repo: this.state.repository
-          };
-          fetch(this.state.git2pantheonURL + "/clone", {
-            body: JSON.stringify(payload),
-            method: "POST"
-          }).then(response => {
-            if (response.status === 201 || response.status === 200) {
-              console.log(" Works " + response.status)
-              this.setState({ isFormSubmitted: true, isSucess: true, msgType: "success", submitMsg: "The git import has been submitted it might take up to 1 minute to see it in the module library" })
-            } else if (response.status === 500) {
+          if (this.state.git2pantheonURL.includes("DOCTYPE html")) {
```

Comment:
Hmm... I'm not sure how I feel about this new technique. It assumes that any error message will be html, and anything else is a success message. I don't think that is a reliable assumption, and I think it may only be true because we wrote the redirect handler (i.e. assume anything that doesn't resolve is a router request like /login or /search). Was relying on the status code not a reliable technique?

Reply:
So if I curl with -v to see the exchange I see that our servlet is always giving us 200 even if the resource does not exist. 

```
curl -v localhost:8080/content/repository/random
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /content/repository/random HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.64.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Date: Mon, 01 Jul 2019 19:44:41 GMT
< X-Content-Type-Options: nosniff
< X-Frame-Options: SAMEORIGIN
< Content-Type: text/html;charset=utf-8
< Transfer-Encoding: chunked
< 
<!DOCTYPE html>
<html lang="en-US">

<head>
  <meta charset="utf-8">
  <title>Pantheon</title>
  <meta id="appName" name="application-name" content="Pantheon">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <link rel="icon" type="image/png" href="https://www.patternfly.org/components/patternfly/dist/img/favicon.ico">
<link href="app.css" rel="stylesheet"></head>

<body>
  <noscript>Enabling JavaScript is required to run this app.</noscript>
  <div id="root"></div>
<script type="text/javascript" src="app.bundle.js"></script></body>

</html>

```

I can change it to do an additional check, if it's not 200 then throw the same Alert. However, for the issue you saw I think checking for HTML is the way to catch it because of what we are doing in the backend.

Reply:
@benradey added some more error handling.